File: frontend/src/lib/languages.ts
export const LANGUAGE_OPTIONS = [
  "English",
  "Arabic",
  "Tagalog",
  "Urdu",
  "Hindi",
] as const;

export type LanguageOption = (typeof LANGUAGE_OPTIONS)[number];

export function normalizeLanguage(value: string) {
  const trimmed = value.trim();
  if (!trimmed) return "";
  const known = LANGUAGE_OPTIONS.find(language => language.toLowerCase() === trimmed.toLowerCase());
  return known || trimmed.replace(/\b\w/g, char => char.toUpperCase());
}

export function normalizeLanguages(values: string[]) {
  const seen = new Set<string>();
  const normalized: string[] = [];
  for (const value of values) {
    const language = normalizeLanguage(value);
    if (!language) continue;
    const key = language.toLowerCase();
    if (seen.has(key)) continue;
    seen.add(key);
    normalized.push(language);
  }
  return normalized;
}

File: api/languages.py
"""Language normalization helpers for user language preferences."""

LANGUAGE_OPTIONS = ["English", "Arabic", "Tagalog", "Urdu", "Hindi"]

_LANGUAGE_LOOKUP = {language.lower(): language for language in LANGUAGE_OPTIONS}


def normalize_languages(values) -> list[str]:
    if not isinstance(values, list):
        raise ValueError("Languages must be a list.")

    normalized: list[str] = []
    seen: set[str] = set()
    for raw in values:
        if raw is None:
            continue
        value = str(raw).strip()
        if not value:
            continue
        canonical = _LANGUAGE_LOOKUP.get(value.lower(), value.title())
        key = canonical.lower()
        if key in seen:
            continue
        seen.add(key)
        normalized.append(canonical)
    return normalized
